tools/secrets/set-secret.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/zsh
+
+# Usage:
+# $ ./tools/secrets/set-secret.sh
+#
+# Note:
+# - Requires `vault` to be installed
+
+set -eo pipefail
+source ./tools/utils/echo-color.sh
+source ./tools/secrets/config.sh
+
+list_secrets() {
+    GREEN="\e[32m"
+    RESET="\e[0m"
+
+    echo "Available secrets:"
+    for key in ${(k)DD_IOS_SECRETS}; do
+        IFS=" | " read -r name description <<< "${DD_IOS_SECRETS[$key]}"
+        echo "$key) ${GREEN}$name${RESET} - $description"
+    done | sort -n
+
+    echo ""
+    echo "To add a new secret, first define it in 'tools/secrets/config.sh' and retry."
+}
+
+select_secret() {
+    echo
+    while true; do
+        echo_info "Enter the number of the secret you want to set:"
+        read "secret_number"
+        if [[ -n ${DD_IOS_SECRETS[$secret_number]} ]]; then
+            IFS=" | " read -r SECRET_NAME SECRET_DESC <<< "${DD_IOS_SECRETS[$secret_number]}"
+            break
+        else
+            echo_err "Invalid selection. Please enter a valid number."
+        fi
+    done
+}
+
+get_secret_value_from_input() {
+    echo_info "Enter the new value for '$SECRET_NAME':"
+    read "SECRET_VALUE"
+    echo
+}
+
+get_secret_value_from_file() {
+    echo_info "Enter the file path to read the value for '$SECRET_NAME':"
+    read "SECRET_FILE"
+    echo
+
+    SECRET_FILE=${SECRET_FILE/#\~/$HOME} # Expand ~ to home directory if present
+    echo_info "Using '$SECRET_FILE'"
+
+    if [[ -f "$SECRET_FILE" ]]; then
+        SECRET_VALUE=$(cat "$SECRET_FILE")
+    else
+        echo_err "Error: File '$SECRET_FILE' does not exist."
+        exit 1
+    fi
+}
+
+select_input_method() {
```

Comment:
how is it going to work in a CI env? How can you enter manually in gitlab CI during the execution?

🤔

Reply:
The `set-secret.sh` flow is made for local use only. We don't **set** secrets from CI - CI only **gets** them (through `get-secret.sh`).